lib/usb-device.js

```diff
@@ -0,0 +1,237 @@
+// import constants
+import { USB_VENDOR_IDS, ADB_VALUES, CONNECT_VALUES,
+          ADB_COMMANDS, ADB_KEY } from './constants';
+let MAXDATA = CONNECT_VALUES.CONNECT_MAXDATA;
+// import helpers
+import { generateMessage, packetFromBuffer } from './helpers';
+
+// required libraries
+import path from 'path';
+import fs from 'fs';
+import Promise from 'bluebird';
+import * as usb from 'usb';
+Promise.promisifyAll(usb);
```

Comment:
Initially, from reading the documentation about promisify, I thought calling promisifyAll on `usb` would promisify functions on the objects within a usb object, such as the endpoints.  Obviously that's not the case, so I can remove the promisify on `usb`.  I can only see one callback function on the usb object I might need to call anyways.